PriceQuoteIos/Utils/BaseCallback.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  BaseCallback.swift
+//  PriceQuoteIos
+//
+//  Created by Livia Vasconcelos on 27/09/20.
+//  Copyright © 2020 Livia Vasconcelos. All rights reserved.
+//
+
+public class BaseCallback<T> {
```

Comment:
Just like you, I really like to use this class, but I would like to provoke you to think a little bit about it: The way that you drawn the project architecture, is this going to be really needed, or it is only used because it is convenient? The issue to always use this, is because we'll be always relying on this structure and we probably won't find this in any other project around!
I guess this could be a good thing to search and study a little about, and test to validate if it makes more sense to keep or change it!

Reply:
Removed.